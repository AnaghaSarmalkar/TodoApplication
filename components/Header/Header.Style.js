import { StyleSheet } from "react-native";

export const style = StyleSheet.create({
  header: {
    fontSize: 32,
    backgroundColor: "#fff",
  },
});
