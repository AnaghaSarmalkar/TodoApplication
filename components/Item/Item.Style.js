import { StyleSheet } from "react-native";

export const style = StyleSheet.create({
  item: {
    padding: 20,
    marginVertical: 8,
    marginHorizontal: 16,
  },
  name: {
    fontSize: 32,
  },
});
