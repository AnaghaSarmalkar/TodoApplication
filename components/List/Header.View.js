import React from "react";
import { StyleSheet, Text, View } from "react-native";

const styles = StyleSheet.create({
  header: {
    fontSize: 32,
    backgroundColor: "#fff",
  },
});

export default function Header({ title }) {
  return (
    <View>
      <Text style={styles.header}>{title}</Text>
    </View>
  );
}
