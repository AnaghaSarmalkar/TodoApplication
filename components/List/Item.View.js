import React from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";

const styles = StyleSheet.create({
  item: {
    padding: 20,
    marginVertical: 8,
    marginHorizontal: 16,
  },
  title: {
    fontSize: 32,
  },
});

export default function Item({ item }) {
  return (
    <TouchableOpacity style={[styles.item]}>
      <Text style={[styles.title]}>{item.title}</Text>
    </TouchableOpacity>
  );
}
