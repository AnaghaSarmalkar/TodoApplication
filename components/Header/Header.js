import React from "react";
import { Text } from "react-native";
import { styles as basicStyles } from "./Header.Style";

export default function Header({ styles = basicStyles, title }) {
  return <Text style={styles.header}>{title}</Text>;
}
