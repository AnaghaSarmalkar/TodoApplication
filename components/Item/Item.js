import React from "react";
import { Text, TouchableOpacity } from "react-native";
import { style } from "./Item.Style";

export default function Item({ item }) {
  return (
    <TouchableOpacity style={[style.item]}>
      <Text style={[style.name]}>{item.name}</Text>
    </TouchableOpacity>
  );
}
