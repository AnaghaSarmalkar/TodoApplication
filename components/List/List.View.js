import React from "react";
import { FlatList, SafeAreaView, StatusBar, StyleSheet } from "react-native";
import Header from "./Header.View";
import Item from "./Item.View";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: StatusBar.currentHeight || 0,
  },
  item: {
    backgroundColor: "#f9c2ff",
    padding: 20,
    marginVertical: 8,
    marginHorizontal: 16,
  },
  title: {
    fontSize: 32,
  },
});

export default function List({ data }) {
  return (
    <SafeAreaView style={styles.container}>
      <FlatList
        ListHeaderComponent={<Header title="Todo List" />}
        data={data}
        renderItem={({ item }) => <Item item={item} />}
        keyExtractor={(item) => item.id}
      />
    </SafeAreaView>
  );
}
