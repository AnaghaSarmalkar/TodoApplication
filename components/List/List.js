import React from "react";
import { FlatList } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import Header from "../Header/Header";
import Item from "../Item/Item";

export default function List({ data }) {
  return (
    <SafeAreaView>
      <FlatList
        ListHeaderComponent={<Header title="Todo List" />}
        data={data}
        renderItem={({ item }) => <Item item={item} />}
        keyExtractor={(item) => item.id}
      />
    </SafeAreaView>
  );
}
