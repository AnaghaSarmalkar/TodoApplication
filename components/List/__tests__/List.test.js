import React from "react";
import { shallow } from "../../../utilities/testing";
import Header from "../../Header/Header";
import Item from "../../Item/Item";
import List from "../List";

describe("", () => {
  it("renders List component", async () => {
    const data = [{ id: "1", name: "Test Item name" }];
    const header = <Header title={"Test Header component."}></Header>;
    const renderItem = ({ item }) => <Item item={item} />;
    const wrapper = shallow(
      <List data={data} headerComponent={header} renderItem={renderItem} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
