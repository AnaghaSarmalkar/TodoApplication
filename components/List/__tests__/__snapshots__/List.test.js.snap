// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders List component 1`] = `
<View>
  <FlatList
    ListHeaderComponent={
      <Header
        title="Test Header component."
      />
    }
    data={
      [
        {
          "id": "1",
          "name": "Test Item name",
        },
      ]
    }
    keyExtractor={[Function]}
    renderItem={[Function]}
  />
</View>
`;
